globals.go

```diff
@@ -538,6 +538,121 @@ const (
 	HostsNone          HostsPermission          = 0
 )
 
+func MapPermissionKeyToName(permissionKey string) string {
+	switch permissionKey {
+	case PermissionProjectSettings:
+		return "Settings"
+	case PermissionTasks:
+		return "Tasks"
+	case PermissionPatches:
+		return "Patches"
+	case PermissionLogs:
+		return "Logs"
+	case PermissionDistroSettings:
+		return "Distro Settings"
+	case PermissionHosts:
+		return "Distro Hosts"
+	}
+	return ""
+}
+
+type Permission struct {
```

Comment:
This data structure is the same as the PermissionLevel interface. This should probably replace that one. You'll need to change a couple of places where it's used, but it shouldn't be too painful (however it's more complicated than doing a find+replace)

Reply:
refactored it